File: app/src/main/java/com/bkjcb/rqapplication/fragment/GasReviewFragment.java
package com.bkjcb.rqapplication.fragment;

import android.content.Intent;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.allen.library.SuperTextView;
import com.bkjcb.rqapplication.Constants;
import com.bkjcb.rqapplication.MediaPlayActivity;
import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.adapter.FileListAdapter;
import com.bkjcb.rqapplication.interfaces.OnPageButtonClickListener;
import com.bkjcb.rqapplication.model.MediaFile;
import com.bkjcb.rqapplication.model.ReviewRecord;
import com.bkjcb.rqapplication.util.Utils;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.hss01248.dialog.StyledDialog;
import com.luck.picture.lib.PictureSelector;
import com.luck.picture.lib.config.PictureConfig;
import com.luck.picture.lib.config.PictureMimeType;
import com.luck.picture.lib.entity.LocalMedia;
import com.wdullaer.materialdatetimepicker.date.DatePickerDialog;

import java.util.Calendar;
import java.util.List;
import java.util.Locale;

import butterknife.BindView;
import butterknife.OnClick;

import static android.app.Activity.RESULT_OK;

/**
 * Created by DengShuai on 2020/4/29.
 * Description :
 */
public class GasReviewFragment extends BaseSimpleFragment implements DatePickerDialog.OnDateSetListener, RadioGroup.OnCheckedChangeListener {
    @BindView(R.id.info_unit)
    EditText mInfoUnit;
    @BindView(R.id.info_station)
    TextView mInfoStation;
    @BindView(R.id.info_name_title)
    TextView mInfoNameTitle;
    @BindView(R.id.check_result_radio_ok)
    RadioButton mCheckResultRadioOk;
    @BindView(R.id.check_result_radio_failure)
    RadioButton mCheckResultRadioFailure;
    @BindView(R.id.check_result_radioGroup)
    RadioGroup mCheckResultRadioGroup;
    @BindView(R.id.check_yhq)
    SuperTextView mCheckYhq;
    @BindView(R.id.check_yhq_time)
    TextView mCheckYhqTime;
    @BindView(R.id.check_yhq_layout)
    LinearLayout mCheckYhqLayout;
    @BindView(R.id.check_tyq_count)
    EditText mCheckTyqCount;
    @BindView(R.id.check_tyq_change)
    SuperTextView mCheckTyqChange;
    @BindView(R.id.check_tyq_time)
    TextView mCheckTyqTime;
    @BindView(R.id.check_tyq_layout)
    LinearLayout mCheckTyqLayout;
    @BindView(R.id.check_ljg_count)
    EditText mCheckLjgCount;
    @BindView(R.id.check_ljg_change)
    SuperTextView mCheckLjgChange;
    @BindView(R.id.check_ljg_time)
    TextView mCheckLjgTime;
    @BindView(R.id.check_ljg_layout)
    LinearLayout mCheckLjgLayout;
    @BindView(R.id.check_rj_count)
    EditText mCheckRjCount;
    @BindView(R.id.check_rj_change)
    SuperTextView mCheckRjChange;
    @BindView(R.id.check_rj_time)
    TextView mCheckRjTime;
    @BindView(R.id.check_rj_layout)
    LinearLayout mCheckRjLayout;
    @BindView(R.id.content_layout)
    LinearLayout mContentLayout;
    @BindView(R.id.record_submit)
    TextView mSubmit;
    @BindView(R.id.file_info)
    RecyclerView mFileInfo;
    @BindView(R.id.file_count)
    TextView mFileCount;
    @BindView(R.id.info_username)
    TextView mUserName;
    private DatePickerDialog pickerDialog;
    private TextView currentTextView;
    private ReviewRecord record;
    private boolean isCanChange = true;
    private OnPageButtonClickListener listener;
    private FileListAdapter imageAdapter;

    public void setRecord(ReviewRecord record) {
        this.record = record;
    }

    public void setListener(OnPageButtonClickListener listener) {
        this.listener = listener;
    }

    @Override
    public void setResId() {
        resId = R.layout.fragment_review_detail;
    }

    public static GasReviewFragment newInstance(ReviewRecord record) {
        GasReviewFragment fragment = new GasReviewFragment();
        fragment.setRecord(record);
        return fragment;
    }

    public static GasReviewFragment newInstance(ReviewRecord record, OnPageButtonClickListener listener) {
        GasReviewFragment fragment = new GasReviewFragment();
        fragment.setRecord(record);
        fragment.setListener(listener);
        return fragment;
    }

    @Override
    protected void initView() {
        isCanChange = record.status == 1;
        setText(mUserName, record.yonghuming);
        if (isCanChange) {
            mCheckYhq.setSwitchCheckedChangeListener(new SuperTextView.OnSwitchCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    setViewVisibility(mContentLayout, !isChecked);
                    setViewVisibility(mCheckYhqLayout, isChecked);
                    record.yehuaqi_shiyong = isChecked ? "是" : "否";
                }
            });
            mCheckTyqChange.setSwitchCheckedChangeListener(new SuperTextView.OnSwitchCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    setViewVisibility(mCheckTyqLayout, isChecked);
                    record.tiaoyaqi_zhenggai = isChecked ? "是" : "否";
                }
            });
            mCheckLjgChange.setSwitchCheckedChangeListener(new SuperTextView.OnSwitchCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    setViewVisibility(mCheckLjgLayout, isChecked);
                    record.lianjieguan_zhenggai = isChecked ? "是" : "否";
                }
            });
            mCheckRjChange.setSwitchCheckedChangeListener(new SuperTextView.OnSwitchCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    setViewVisibility(mCheckRjLayout, isChecked);
                    record.ranju_zhenggai = isChecked ? "是" : "否";
                }
            });
        } else {
            setText(mInfoUnit, record.jianchadanwei);
            setText(mInfoStation, record.jianchariqi);
            mCheckResultRadioOk.setEnabled(false);
            mCheckResultRadioFailure.setEnabled(false);
            setChecked(mCheckYhq, "是".equals(record.yehuaqi_shiyong), mCheckYhqLayout, mCheckYhqTime, record.tuihuriqi);
            if ("是".equals(record.yehuaqi_shiyong)) {
                setViewVisibility(mContentLayout, false);
            }
            setText(mCheckTyqCount, record.tiaoyaqi_geshu);
            setChecked(mCheckTyqChange, "是".equals(record.tiaoyaqi_zhenggai), mCheckTyqLayout, mCheckTyqTime, record.tiaoyaqi_zhenggairiqi);
            setText(mCheckLjgCount, record.lianjieguan_geshu);
            setChecked(mCheckLjgChange, "是".equals(record.lianjieguan_zhenggai), mCheckLjgLayout, mCheckLjgTime, record.lianjieguan_zhenggairiqi);
            setText(mCheckRjCount, record.ranju_geshu);
            setChecked(mCheckRjChange, "是".equals(record.ranju_zhenggai), mCheckRjLayout, mCheckRjTime, record.ranju_zhenggairiqi);
        }
        initCheckBox();
    }

    @Override
    protected void initData() {
        if (isCanChange) {
            StyledDialog.init(getContext());
            record.tiaoyaqi_zhenggai = "否";
            record.ranju_zhenggai = "否";
            record.yehuaqi_shiyong = "否";
            record.lianjieguan_zhenggai = "否";
        } else {
            setViewVisibility(mSubmit, false);
        }
        initFileView();
    }

    @OnClick({R.id.info_station, R.id.check_yhq_time, R.id.check_tyq_time, R.id.check_ljg_time, R.id.check_rj_time, R.id.record_submit})
    public void onClick(View v) {
        if (v.getId() != R.id.record_submit) {
            switch (v.getId()) {
                case R.id.info_station:
                    currentTextView = mInfoStation;
                    break;
                case R.id.check_yhq_time:
                    currentTextView = mCheckYhqTime;
                    break;
                case R.id.check_tyq_time:
                    currentTextView = mCheckTyqTime;
                    break;
                case R.id.check_ljg_time:
                    currentTextView = mCheckLjgTime;
                    break;
                case R.id.check_rj_time:
                    currentTextView = mCheckRjTime;
                    break;
                default:
                    break;
            }
            createDatePicker();
        } else {
            collectParams();
            if (verifyAllData()) {
                listener.onNext(imageAdapter.getData());
            }
        }
    }

    private void setViewVisibility(View v, boolean isShow) {
        v.setVisibility(isShow ? View.VISIBLE : View.GONE);
    }

    private void createDatePicker() {
        if (pickerDialog == null) {
            Calendar calendar = Utils.getCalendar();
            pickerDialog = DatePickerDialog.newInstance(
                    this,
                    calendar.get(Calendar.YEAR), // Initial year selection
                    calendar.get(Calendar.MONTH), // Initial month selection
                    calendar.get(Calendar.DAY_OF_MONTH) // Inital day selection
            );
        }
        if (!pickerDialog.isAdded()) {
            pickerDialog.show(getActivity().getFragmentManager(), "DatePickerDialog");
        }
    }

    @Override
    public void onDateSet(DatePickerDialog view, int year, int monthOfYear, int dayOfMonth) {
        currentTextView.setText(String.format(Locale.CHINESE, "%s-%s-%s", year, monthOfYear + 1, dayOfMonth));
    }

    private void initCheckBox() {
        if (!TextUtils.isEmpty(record.jianchajieguo)) {
            if ("合格".equals(record.jianchajieguo)) {
                mCheckResultRadioGroup.check(R.id.check_result_radio_ok);
                changeTextColor(true);
            } else {
                mCheckResultRadioGroup.check(R.id.check_result_radio_failure);
                changeTextColor(false);
            }
        } else {
            mCheckResultRadioGroup.setOnCheckedChangeListener(this);
        }
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        if (checkedId == R.id.check_result_radio_ok) {
            record.jianchajieguo = "合格";
            changeTextColor(true);
        } else {
            record.jianchajieguo = "不合格";
            changeTextColor(false);
        }
    }

    private void changeTextColor(boolean isOk) {
        if (isOk) {
            mCheckResultRadioOk.setTextColor(getResources().getColor(R.color.colorText));
            mCheckResultRadioFailure.setTextColor(getResources().getColor(R.color.colorSecondDrayText));
        } else {
            mCheckResultRadioOk.setTextColor(getResources().getColor(R.color.colorSecondDrayText));
            mCheckResultRadioFailure.setTextColor(getResources().getColor(R.color.colorText));
        }
    }

    private void setText(TextView view, String value) {
        view.setText(value);
        view.setEnabled(false);
    }

    private void setChecked(SuperTextView box, boolean enable, View layout, TextView view, String value) {
        box.setSwitchIsChecked(enable);
        box.setSwitchClickable(false);
        setViewVisibility(layout, enable);
        if (enable && view != null) {
            setText(view, value);
        }
    }

    private String getText(TextView view) {
        return view.getText().toString();
    }

    private boolean verifyAllData() {
        return verify(mInfoUnit, record.jianchadanwei, "请填写检查单位") &&
                verify(mInfoStation, record.jianchariqi, "请选择检查日期") &&
                verify(mInfoNameTitle, record.jianchajieguo, "请选择检查结果") &&
                verify(mFileInfo, "请添加三到五张照片") &&
                (("是".equals(record.yehuaqi_shiyong) &&
                        verify(mCheckYhqTime, record.tuihuriqi, "请选择退户日期")
                ) || ("否".equals(record.yehuaqi_shiyong) &&
                        verify(mCheckTyqCount, record.tiaoyaqi_geshu, "请填写已置换调压器个数")
                        &&
                        (("是".equals(record.tiaoyaqi_zhenggai) &&
                                verify(mCheckTyqTime, record.tiaoyaqi_zhenggairiqi, "请选择整改日期")
                        ) || ("否".equals(record.tiaoyaqi_zhenggai)))
                        &&
                        verify(mCheckLjgCount, record.lianjieguan_geshu, "请填写已置换连接管个数")
                        &&
                        (("是".equals(record.lianjieguan_zhenggai) &&
                                verify(mCheckLjgTime, record.lianjieguan_zhenggairiqi, "请选择整改日期")
                        ) || ("否".equals(record.lianjieguan_zhenggai)))
                        &&
                        verify(mCheckRjCount, record.ranju_geshu, "请填写已置换调压器个数") &&
                        (("是".equals(record.ranju_zhenggai) &&
                                verify(mCheckRjTime, record.ranju_zhenggairiqi, "请选择整改日期")
                        ) || ("否".equals(record.ranju_zhenggai)))))
                ;
    }

    private boolean verify(View view, String value, String tip) {
        if (TextUtils.isEmpty(value)) {
            view.requestFocus();
            Toast.makeText(getContext(), tip, Toast.LENGTH_SHORT).show();
            return false;
        }
        return true;
    }

    private boolean verify(View view, String tip) {
        int size = imageAdapter.getData().size();
        if (size < 3 || size > 6) {
            view.requestFocus();
            Toast.makeText(getContext(), tip, Toast.LENGTH_SHORT).show();
            return false;
        }
        return true;
    }

    private void collectParams() {
        record.jianchariqi = getText(mInfoStation);
        record.jianchadanwei = getText(mInfoUnit);
        record.tiaoyaqi_geshu = getText(mCheckTyqCount);
        record.tiaoyaqi_zhenggairiqi = getText(mCheckTyqTime);
        record.lianjieguan_geshu = getText(mCheckLjgCount);
        record.lianjieguan_zhenggairiqi = getText(mCheckLjgTime);
        record.ranju_geshu = getText(mCheckRjCount);
        record.ranju_zhenggairiqi = getText(mCheckRjTime);
        record.tuihuriqi = getText(mCheckYhqTime);
    }

    private View createFooterView() {
        int width = Utils.dip2px(getContext(), 120);
        ImageView view = new ImageView(getContext());
        view.setLayoutParams(new ViewGroup.LayoutParams(width, width));
        int padding = Utils.dip2px(getContext(), 5);
        view.setPadding(padding, padding, padding, padding);
        view.setImageResource(R.drawable.icon_add_pic);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showPickImg();
            }
        });
        return view;
    }

    private void initFileView() {

        mFileInfo.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false));
        imageAdapter = new FileListAdapter(R.layout.item_image, isCanChange);
        mFileInfo.setAdapter(imageAdapter);
        if (isCanChange) {
            imageAdapter.setFooterView(createFooterView());
        }
        imageAdapter.setOnItemClickListener(new BaseQuickAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
                MediaPlayActivity.ToActivity(getContext(), ((MediaFile) adapter.getItem(position)).getPath());
            }
        });
        imageAdapter.setOnItemChildClickListener(new BaseQuickAdapter.OnItemChildClickListener() {
            @Override
            public void onItemChildClick(BaseQuickAdapter adapter, View view, int position) {
                if (view.getId() == R.id.item_grid_bt) {
                    imageAdapter.remove(position);
                    refreshFileCount();
                }
            }
        });
        if (!TextUtils.isEmpty(record.phoneftp)) {
            String[] paths = record.phoneftp.split(",");
            for (String path : paths) {
                MediaFile mediaFile = new MediaFile();
                mediaFile.setPath(Constants.IMAGE_URL + path);
                mediaFile.setLocal(false);
                imageAdapter.addData(mediaFile);
            }
        }
        refreshFileCount();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (resultCode == RESULT_OK && requestCode == PictureConfig.CHOOSE_REQUEST) {
            List<LocalMedia> list = PictureSelector.obtainMultipleResult(data);
            handleMedia(list);
        }
    }

    private void handleMedia(List<LocalMedia> list) {
        MediaFile mediaFile;
        for (LocalMedia media : list) {
            mediaFile = new MediaFile();
            mediaFile.setType(media.getMimeType());
            mediaFile.setPath(media.getCompressPath());
            imageAdapter.addData(mediaFile);
        }
        refreshFileCount();
    }

    private void refreshFileCount() {
        mFileCount.setText(String.valueOf(imageAdapter.getData().size()));
    }

    public void showPickImg() {
        PictureSelector.create(this)
                .openGallery(PictureConfig.TYPE_IMAGE)
                .setOutputCameraPath("/RQApp/GasImage/")
                .compress(true)
          /*      .compressSavePath(Environment.getExternalStorageDirectory()
                        + "/RQApp/GasImage/compress")*/
                .minimumCompressSize(300)
                .imageFormat(PictureMimeType.PNG)
                //.selectionMedia(selectList)
                .forResult(PictureConfig.CHOOSE_REQUEST);
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/gasrecord/fragment/GasReviewFragment.kt
package com.bkjcb.rqapplication.gasrecord.fragment

import android.app.Activity
import android.content.Intent
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.View
import android.view.ViewGroup
import android.widget.*
import butterknife.BindView
import butterknife.OnClick
import com.allen.library.SuperTextView
import com.bkjcb.rqapplication.Constants
import com.bkjcb.rqapplication.MediaPlayActivity
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.adapter.FileListAdapter
import com.bkjcb.rqapplication.fragment.BaseSimpleFragment
import com.bkjcb.rqapplication.gasrecord.model.ReviewRecord
import com.bkjcb.rqapplication.interfaces.OnPageButtonClickListener
import com.bkjcb.rqapplication.model.MediaFile
import com.bkjcb.rqapplication.util.PictureSelectorUtil
import com.bkjcb.rqapplication.util.Utils
import com.bkjcb.rqapplication.view.FooterView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.hss01248.dialog.StyledDialog
import com.luck.picture.lib.PictureSelector
import com.luck.picture.lib.config.PictureConfig
import com.luck.picture.lib.config.PictureMimeType
import com.luck.picture.lib.entity.LocalMedia
import com.wdullaer.materialdatetimepicker.date.DatePickerDialog
import kotlinx.android.synthetic.main.fragment_review_detail.*
import java.util.*

/**
 * Created by DengShuai on 2020/4/29.
 * Description :
 */
class GasReviewFragment : BaseSimpleFragment(), DatePickerDialog.OnDateSetListener, RadioGroup.OnCheckedChangeListener, View.OnClickListener {


    private var pickerDialog: DatePickerDialog? = null
    private var currentTextView: TextView? = null
    private var record: ReviewRecord? = null
    private var isCanChange = true
    private var listener: OnPageButtonClickListener<*>? = null
    private lateinit var imageAdapter: FileListAdapter
    fun setRecord(record: ReviewRecord?) {
        this.record = record
    }

    fun setListener(listener: OnPageButtonClickListener<*>?) {
        this.listener = listener
    }

    override fun initView() {
        isCanChange = record!!.status == 1
        setText(info_username, record!!.yonghuming)
        if (isCanChange) {
            check_yhq.setSwitchCheckedChangeListener { buttonView, isChecked ->
                setViewVisibility(content_layout, !isChecked)
                setViewVisibility(check_yhq_layout, isChecked)
                record!!.yehuaqi_shiyong = if (isChecked) "是" else "否"
            }
            check_tyq_change.setSwitchCheckedChangeListener { buttonView, isChecked ->
                setViewVisibility(check_tyq_layout, isChecked)
                record!!.tiaoyaqi_zhenggai = if (isChecked) "是" else "否"
            }
            check_ljg_change.setSwitchCheckedChangeListener { buttonView, isChecked ->
                setViewVisibility(check_ljg_layout, isChecked)
                record!!.lianjieguan_zhenggai = if (isChecked) "是" else "否"
            }
            check_rj_change.setSwitchCheckedChangeListener { buttonView, isChecked ->
                setViewVisibility(check_rj_layout, isChecked)
                record!!.ranju_zhenggai = if (isChecked) "是" else "否"
            }
            //@OnClick(R.id.info_station, R.id.check_yhq_time, R.id.check_tyq_time, R.id.check_ljg_time, R.id.check_rj_time, R.id.record_submit)
            info_station.setOnClickListener(this)
            check_yhq_time.setOnClickListener(this)
            check_tyq_time.setOnClickListener(this)
            check_ljg_time.setOnClickListener(this)
            check_rj_time.setOnClickListener(this)
            record_submit.setOnClickListener(this)
        } else {
            setText(info_unit, record!!.jianchadanwei)
            setText(info_station, record!!.jianchariqi)
            check_result_radio_ok.isEnabled = false
            check_result_radio_failure.isEnabled = false
            setChecked(check_yhq, "是" == record!!.yehuaqi_shiyong, check_yhq_layout, check_yhq_time, record!!.tuihuriqi)
            if ("是" == record!!.yehuaqi_shiyong) {
                setViewVisibility(content_layout, false)
            }
            setText(check_tyq_count, record!!.tiaoyaqi_geshu)
            setChecked(check_tyq_change, "是" == record!!.tiaoyaqi_zhenggai, check_tyq_layout, check_tyq_time, record!!.tiaoyaqi_zhenggairiqi)
            setText(check_ljg_count, record!!.lianjieguan_geshu)
            setChecked(check_ljg_change, "是" == record!!.lianjieguan_zhenggai, check_ljg_layout, check_ljg_time, record!!.lianjieguan_zhenggairiqi)
            setText(check_rj_count, record!!.ranju_geshu)
            setChecked(check_rj_change, "是" == record!!.ranju_zhenggai, check_rj_layout, check_rj_time, record!!.ranju_zhenggairiqi)
        }
        initCheckBox()
    }

    override fun initData() {
        if (isCanChange) {
            StyledDialog.init(context)
            record!!.tiaoyaqi_zhenggai = "否"
            record!!.ranju_zhenggai = "否"
            record!!.yehuaqi_shiyong = "否"
            record!!.lianjieguan_zhenggai = "否"
        } else {
            setViewVisibility(record_submit, false)
        }
        initFileView()
    }


  override  fun onClick(v: View) {
        if (v.id != R.id.record_submit) {
            when (v.id) {
                R.id.info_station -> currentTextView = info_station
                R.id.check_yhq_time -> currentTextView = check_yhq_time
                R.id.check_tyq_time -> currentTextView = check_tyq_time
                R.id.check_ljg_time -> currentTextView = check_ljg_time
                R.id.check_rj_time -> currentTextView = check_rj_time
                else -> {
                }
            }
            createDatePicker()
        } else {
            collectParams()
            if (verifyAllData()) {
                listener!!.onNext(imageAdapter.data)
            }
        }
    }

    private fun setViewVisibility(v: View?, isShow: Boolean) {
        v!!.visibility = if (isShow) View.VISIBLE else View.GONE
    }

    private fun createDatePicker() {
        if (pickerDialog == null) {
            val calendar = Utils.obtainCalendar()
            pickerDialog = DatePickerDialog.newInstance(
                    this,
                    calendar[Calendar.YEAR],  // Initial year selection
                    calendar[Calendar.MONTH],  // Initial month selection
                    calendar[Calendar.DAY_OF_MONTH] // Inital day selection
            )
        }
        if (!pickerDialog!!.isAdded) {
            pickerDialog!!.show(activity!!.fragmentManager, "DatePickerDialog")
        }
    }

    override fun onDateSet(view: DatePickerDialog, year: Int, monthOfYear: Int, dayOfMonth: Int) {
        currentTextView!!.text = String.format(Locale.CHINESE, "%s-%s-%s", year, monthOfYear + 1, dayOfMonth)
    }

    private fun initCheckBox() {
        if (!TextUtils.isEmpty(record!!.jianchajieguo)) {
            if ("合格" == record!!.jianchajieguo) {
                check_result_radioGroup.check(R.id.check_result_radio_ok)
                changeTextColor(true)
            } else {
                check_result_radioGroup.check(R.id.check_result_radio_failure)
                changeTextColor(false)
            }
        } else {
            check_result_radioGroup.setOnCheckedChangeListener(this)
        }
    }

    override fun onCheckedChanged(group: RadioGroup, checkedId: Int) {
        if (checkedId == R.id.check_result_radio_ok) {
            record!!.jianchajieguo = "合格"
            changeTextColor(true)
        } else {
            record!!.jianchajieguo = "不合格"
            changeTextColor(false)
        }
    }

    private fun changeTextColor(isOk: Boolean) {
        if (isOk) {
            check_result_radio_ok.setTextColor(resources.getColor(R.color.colorText))
            check_result_radio_failure.setTextColor(resources.getColor(R.color.colorSecondDrayText))
        } else {
            check_result_radio_ok.setTextColor(resources.getColor(R.color.colorSecondDrayText))
            check_result_radio_failure.setTextColor(resources.getColor(R.color.colorText))
        }
    }

    private fun setText(view: TextView?, value: String) {
        view!!.text = value
        view.isEnabled = false
    }

    private fun setChecked(box: SuperTextView?, enable: Boolean, layout: View?, view: TextView?, value: String) {
        box!!.switchIsChecked = enable
        box.setSwitchClickable(false)
        setViewVisibility(layout, enable)
        if (enable && view != null) {
            setText(view, value)
        }
    }

    private fun getText(view: TextView?): String {
        return view!!.text.toString()
    }

    private fun verifyAllData(): Boolean {
        return verify(info_unit, record!!.jianchadanwei, "请填写检查单位") &&
                verify(info_station, record!!.jianchariqi, "请选择检查日期") &&
                verify(info_name_title, record!!.jianchajieguo, "请选择检查结果") &&
                verify(file_info, "请添加三到五张照片") &&
                ("是" == record!!.yehuaqi_shiyong &&
                        verify(check_yhq_time, record!!.tuihuriqi, "请选择退户日期") || ("否" == record!!.yehuaqi_shiyong &&
                        verify(check_tyq_count, record!!.tiaoyaqi_geshu, "请填写已置换调压器个数")
                        &&
                        ("是" == record!!.tiaoyaqi_zhenggai &&
                                verify(check_tyq_time, record!!.tiaoyaqi_zhenggairiqi, "请选择整改日期") || "否" == record!!.tiaoyaqi_zhenggai)
                        &&
                        verify(check_ljg_count, record!!.lianjieguan_geshu, "请填写已置换连接管个数")
                        &&
                        ("是" == record!!.lianjieguan_zhenggai &&
                                verify(check_ljg_time, record!!.lianjieguan_zhenggairiqi, "请选择整改日期") || "否" == record!!.lianjieguan_zhenggai)
                        &&
                        verify(check_rj_count, record!!.ranju_geshu, "请填写已置换调压器个数") &&
                        ("是" == record!!.ranju_zhenggai &&
                                verify(check_rj_time, record!!.ranju_zhenggairiqi, "请选择整改日期") || "否" == record!!.ranju_zhenggai)))
    }

    private fun verify(view: View?, value: String, tip: String): Boolean {
        if (TextUtils.isEmpty(value)) {
            view!!.requestFocus()
            Toast.makeText(context, tip, Toast.LENGTH_SHORT).show()
            return false
        }
        return true
    }

    private fun verify(view: View?, tip: String): Boolean {
        val size = imageAdapter.data.size
        if (size < 3 || size > 6) {
            view!!.requestFocus()
            Toast.makeText(context, tip, Toast.LENGTH_SHORT).show()
            return false
        }
        return true
    }

    private fun collectParams() {
        record!!.jianchariqi = getText(info_station)
        record!!.jianchadanwei = getText(info_unit)
        record!!.tiaoyaqi_geshu = getText(check_tyq_count)
        record!!.tiaoyaqi_zhenggairiqi = getText(check_tyq_time)
        record!!.lianjieguan_geshu = getText(check_ljg_count)
        record!!.lianjieguan_zhenggairiqi = getText(check_ljg_time)
        record!!.ranju_geshu = getText(check_rj_count)
        record!!.ranju_zhenggairiqi = getText(check_rj_time)
        record!!.tuihuriqi = getText(check_yhq_time)
    }

    private fun createFooterView(): View {
        return FooterView.createFooter(View.OnClickListener {
            showPickImg()
        })
    }

    private fun initFileView() {
        file_info.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
        imageAdapter = FileListAdapter(R.layout.item_image, isCanChange)
        file_info.adapter = imageAdapter
        if (isCanChange) {
            imageAdapter.setFooterView(createFooterView(), 0, LinearLayout.HORIZONTAL)
        }
        imageAdapter.onItemClickListener = BaseQuickAdapter.OnItemClickListener { adapter, view, position -> MediaPlayActivity.ToActivity(context, (adapter.getItem(position) as MediaFile?)!!.path) }
        imageAdapter.onItemChildClickListener = BaseQuickAdapter.OnItemChildClickListener { adapter, view, position ->
            if (view.id == R.id.item_grid_bt) {
                imageAdapter.remove(position)
                refreshFileCount()
            }
        }
        if (!TextUtils.isEmpty(record!!.phoneftp)) {
            val paths = record!!.phoneftp.split(",".toRegex()).toTypedArray()
            for (path in paths) {
                val mediaFile = MediaFile()
                mediaFile.path = Constants.IMAGE_URL + path
                mediaFile.isLocal = false
                imageAdapter.addData(mediaFile)
            }
        }
        refreshFileCount()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        if (resultCode == Activity.RESULT_OK && requestCode == PictureConfig.CHOOSE_REQUEST) {
            val list = PictureSelector.obtainMultipleResult(data)
            handleMedia(list)
        }
    }

    private fun handleMedia(list: List<LocalMedia>) {
        var mediaFile: MediaFile
        for (media in list) {
            mediaFile = MediaFile()
            mediaFile.type = media.mimeType
            mediaFile.path = media.compressPath
            imageAdapter.addData(mediaFile)
        }
        refreshFileCount()
    }

    private fun refreshFileCount() {
        file_count.text = imageAdapter.data.size.toString()
    }

    fun showPickImg() {
       PictureSelectorUtil.selectPicture(this,"GasImage",true)
    }

    override fun initResID(): Int {
        return R.layout.fragment_review_detail
    }

    companion object {
        fun newInstance(record: ReviewRecord?): GasReviewFragment {
            val fragment = GasReviewFragment()
            fragment.setRecord(record)
            return fragment
        }

        fun newInstance(record: ReviewRecord?, listener: OnPageButtonClickListener<*>?): GasReviewFragment {
            val fragment = GasReviewFragment()
            fragment.setRecord(record)
            fragment.setListener(listener)
            return fragment
        }
    }
}